amy/extrequests/templatetags/request_membership.py

```diff
@@ -0,0 +1,34 @@
+from datetime import date
+
+from django import template
+from django.utils.safestring import mark_safe
+
+from workshops.models import Membership
+
+register = template.Library()
+
+
+@register.simple_tag
+def membership_description(membership: Membership):
+    if not isinstance(membership, Membership):
+        return ""
+
+    workshops_remaining = membership.workshops_without_admin_fee_remaining
+    active = membership.active_on_date(date.today())
+    if workshops_remaining <= 0 or not active:
+        alert_type = "warning"
+    else:
+        alert_type = "info"
+    info = (
+        f'<div class="alert alert-{alert_type}">'
+        "Related membership: "
+        f'<a href="{membership.get_absolute_url()}">{membership}</a>.'
+        "<br>"
+        f"This membership has <strong>{workshops_remaining}</strong> "
+        f'workshop{"s" if workshops_remaining !=1 else ""} remaining.'
```

Comment:
In the Django template you can use [`pluralize`](https://docs.djangoproject.com/en/4.2/ref/templates/builtins/#pluralize) template filter to achieve the plural form of "workshop".